dash-pipeline/bmv2/dash_headers.p4

```diff
@@ -81,6 +85,7 @@ const bit<16> IPV6_HDR_SIZE=320/8;
 struct headers_t {
     ethernet_t ethernet;
     ipv4_t     ipv4;
+    ipv4options_t ipv4options;
```

Comment:
Don't we need to move ipv4options_t part of ipv4_t itself?

Reply:
@saad-mzhr Could you cite the example you based this on from the P4-16 lang spec? You can post the URL to the appropriate line or section in the HTML version of the spec. Thanks.

Reply:
Maybe. I actually adapted this code from p4 official docs on how to handle variable sized headers:
https://p4.org/p4-spec/docs/P4-16-v1.0.0-spec.html#sec-packet-extract-two

Reply:
Personally, I like this approach of treating the extensions header separately, since it's an optional part of the header stack.

Reply:
IPv4 header is basically including options, so, I suggest to rename ipv4_t with IPv4_no_options_t